code/modules/mob/living/silicon/pai/pai.dm

```diff
@@ -406,3 +409,36 @@
 	grabber.update_inv_l_hand()
 	grabber.update_inv_r_hand()
 	return H
+
+/mob/living/silicon/pai/attackby(obj/item/weapon/W as obj, mob/user as mob)
+
+	if (istype(W, /obj/item/weapon/card/id) && idaccessible == 1)

```

Comment:
Additionally, having `istype(W, /obj/item/weapon/card/id)` before `GetID()` prevents PDAs and wallets from working.